### ISSUE TITLE
Right shift and digit 4 press bug on Windows Desktop

### ISSUE THREAD

**Issue by Abduraimbek** [flutter/flutter#110852]
**Appeared on Flutter 3.3.0. I did not have such errors on Flutter 3.0.5.**

The following assertion was thrown during a platform message callback:
```
A KeyRepeatEvent is dispatched, but the state shows that the physical key is not pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyRepeatEvent#c2cad(physicalKey: PhysicalKeyboardKey#700e5(usbHidUsage: "0x000700e5", debugName: "Shift Right"), logicalKey: LogicalKeyboardKey#00103(keyId: "0x200000103", keyLabel: "Shift Right", debugName: "Shift Right"), character: null, timeStamp: 100:09:08.611970)
'package:flutter/src/services/hardware_keyboard.dart':
Failed assertion: line 432 pos 16: '_pressedKeys.containsKey(event.physicalKey)'
```

**Comment by n7484443** [comment]
I think this is reoccurence of the previous  https://github.com/flutter/flutter/issues/108646 error.

**Comment by Spitskool42** [comment]
Hey, same error when using the left shift key in windows 10.

```
════════ Exception caught by services library ══════════════════════════════════
A KeyRepeatEvent is dispatched, but the state shows that the physical key is not pressed. If this occurs in real application, please report this bug to Flutter. If this occurs in unit tests, please ensure that simulated events follow Flutter's event model as documented in `HardwareKeyboard`. This was the event: KeyRepeatEvent#c6ca6(physicalKey: PhysicalKeyboardKey#700e1(usbHidUsage: "0x000700e1", debugName: "Shift Left"), logicalKey: LogicalKeyboardKey#00102(keyId: "0x200000102", keyLabel: "Shift Left", debugName: "Shift Left"), character: null, timeStamp: 483:49:49.697664)
'package:flutter/src/services/hardware_keyboard.dart':
package:flutter/…/services/hardware_keyboard.dart:1
Failed assertion: line 432 pos 16: '_pressedKeys.containsKey(event.physicalKey)'

════════════════════════════════════════════════════════════════════════════════
```

How to recreate the bug in Flutter 3.3.0. 
1. Create a textfield
2. Put text in the textfield
3. Select a part of the text with the mouse
4. Now shift+click somewhere else on the text (to select more text).
5. bug
6. Now backspace for example also does not work anymore (error keeps popping up) in the text untill app restart/reset.

```
[flutter] flutter doctor -v
[√] Flutter (Channel stable, 3.3.0, on Microsoft Windows [Version 10.0.19043.1889], locale en-GB)
    • Flutter version 3.3.0 on channel stable at C:\Users\Gebruiker\flutter\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffccd96b62 (5 days ago), 2022-08-29 17:28:57 -0700
    • Engine revision 5e9e0e0aa8
    • Dart version 2.18.0
    • DevTools version 2.15.0

[√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at C:\Users\Gebruiker\AppData\Local\Android\sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.6)
    • Visual Studio at D:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32630.192
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] VS Code (version 1.71.0)
    • VS Code at C:\Users\Gebruiker\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.48.0

[√] Connected device (2 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1889]
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 104.0.1293.70

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
exit code 0
```

**Comment by Spitskool42** [comment]
Stack trace:
```
 When the exception was thrown, this was the stack:
           #2      HardwareKeyboard._assertEventIsRegular.<anonymous closure>
           (package:flutter/src/services/hardware_keyboard.dart:432:16)
           #3      HardwareKeyboard._assertEventIsRegular (package:flutter/src/services/hardware_keyboard.dart:443:6)
           #4      HardwareKeyboard.handleKeyEvent (package:flutter/src/services/hardware_keyboard.dart:535:5)
           #5      KeyEventManager.handleRawKeyMessage (package:flutter/src/services/hardware_keyboard.dart:881:37)
           #6      BasicMessageChannel.setMessageHandler.<anonymous closure>
           (package:flutter/src/services/platform_channel.dart:197:49)
           #7      _DefaultBinaryMessenger.setMessageHandler.<anonymous closure>
(package:flutter/src/services/binding.dart:387:35)
           #8      _invoke2 (dart:ui/hooks.dart:186:13)
           #9      _ChannelCallbackRecord.invoke (dart:ui/channel_buffers.dart:42:5)
           #10     _Channel.push (dart:ui/channel_buffers.dart:132:31)
           #11     ChannelBuffers.push (dart:ui/channel_buffers.dart:329:17)
           #12     PlatformDispatcher._dispatchPlatformMessage (dart:ui/platform_dispatcher.dart:599:22)
           #13     _dispatchPlatformMessage (dart:ui/hooks.dart:89:31)
           (elided 2 frames from class _AssertionError)
```

**Comment by Spitskool42** [comment]
> I think this is reoccurence of the previous #108646 error.

Hey, I looked this one up and saw that this commit merged into flutter 3.4 candidate 0.
https://github.com/flutter/engine/blob/flutter-3.4-candidate.0/shell/platform/windows/keyboard_key_embedder_handler.cc

And not into the last candidate(12) for flutter 3.3:
https://github.com/flutter/engine/blob/flutter-3.3-candidate.12/shell/platform/windows/keyboard_key_embedder_handler.cc

Does this mean that we need to wait until flutter 3.4 and it will be fixed?

**Comment by huycozy** [comment]
Hi everyone
As @n7484443 and @Spitskool42 mentioned above, there is a similar issue that was fixed recently #108646.

@Abduraimbek Please try the issue again on the latest master channel and confirm if this issue is fixed.
```
flutter channel master
flutter upgrade
```

@Spitskool42 The fix should land on the stable channel in future releases.

**Comment by Spitskool42** [comment]
@huycozy 

Hey, can confirm that this is till not fixed in release 3.3.1 of flutter.
Tested today with flutter 3.3.1.

Tested with channel master, problem was fixed.

**Comment by Abduraimbek** [comment]
I tested on version 3.3.2 and the bug fixed. Awesome!

**Comment by Spitskool42** [comment]
My problem with the shift key + mouseclick to select text is still not fixed in release 3.3.2.
